File: week.3/code/get_TreeHeight.py
"""get_TreeHeight is a python script that calculates tree heights"""

__appname__ = 'get_TreeHeight.py' 
__author__ = 'Peter (Guancheng) Zeng (pz221@ic.ac.uk), Kate Griffin (kate.griffin21@imperial.ac.uk), Junyue Zhang (jz1621@ic.ac.uk), Kayleigh Greenwood (Kayleigh.Greenwood21@ic.ac.uk)'
__version__ = '0.0.1'


### import ###
import sys
import pandas
import numpy

treeData = pandas.read_csv("../data/" + sys.argv[1] + ".csv")

### to calculate the height for every rows ###
for row in treeData:
    treeData["Tree.Height.m"] = treeData["Distance.m"] * numpy.tan(treeData["Angle.degrees"] * numpy.pi/180)

### output the data to the tree data csv
treeData.to_csv("../results/" + sys.argv[1] + "_TreeHts_py.csv", sep= ',',index=False)

File: week.3/code/get_TreeHeight.R
#This function calculates heights of trees given distance of each tree
#from its base and angle to its top, using the trigonometric formula

#height = distance * tan(radians)

#ARGUMENTS
#degrees: The angle of elevation of tree
#distance: The distance from base of tree (e.g., meters)

#OUTPUT
#The heights of the tree, same units as "distance"

rm(list = ls())

args<-commandArgs(TRUE)

### Read the csv file ###
treeData<- read.csv(paste0("../data/",args, ".csv"), stringsAsFactors = F)

TreeHeight <- function(degrees, distance){
  radians <- degrees * pi / 180
  height <- distance * tan(radians)
  return (height)
}


Tree.Height.m <- c()

### Calculate the tree height ###
for (i in 1:nrow(treeData)){
  Tree.Height.m <- append(Tree.Height.m,TreeHeight(treeData$Angle.degrees[i],treeData$Distance.m[i]))
}

treeData$Tree.Height.m <- Tree.Height.m

### Output the result to the csv file ###
write.csv(treeData, paste("../results/",args,"_TreeHts_R.csv",sep = ""),row.names = F)

